### PR TITLE
User Management - Signup / Signin / Changes by Admin

### PR DESCRIPTION
**Pull request by pikender**
Allow Users to Signup and Signin to perform Order Checkout
Admin can CRUD users

### CODE REVIEW

File: apps/ex_shop/web/models/user.ex [hunk 1]
```diff
@@ -6,37 +6,49 @@ defmodule ExShop.User do
...
+    changeset
     |> validate_confirmation(:password, message: "password does not match")
     |> validate_format(:email, ~r/@/)
     |> validate_length(:password, min: 5)
     |> unique_constraint(:email)
+    |> set_hashed_password
+  end
+
+  defp set_hashed_password(changeset = %{errors: [_]}), do: changeset
+  defp set_hashed_password(changeset = %{params: %{"password" => password}}) when password != "" and password != nil do
```
Comment (nimish-mehta) [inline apps/ex_shop/web/models/user.ex]: we can use is_nil for nil comparison: http://elixir-lang.org/docs/stable/elixir/Kernel.html#is_nil/1

File: apps/ex_shop/web/models/user.ex [hunk 2]
```diff
@@ -6,37 +6,49 @@ defmodule ExShop.User do
...
-
-  If no params are provided, an invalid changeset is returned
-  with no validation performed.
+  Refer Registration changeset
   """
-  def register_admin_form_changeset(model, params \\ :empty) do
-    model
-    |> cast(params, ~w(name email password is_admin), ~w())
-  end
+  def create_changeset(model, params \\ :empty), do: changeset(model, params) |> add_error(:email, "Refer Registration changeset")
```
Comment (nimish-mehta) [inline apps/ex_shop/web/models/user.ex:23]: why not remove this changeset, since we cannot use it.

File: apps/ex_shop/web/models/user.ex [hunk 3]
```diff
@@ -6,37 +6,49 @@ defmodule ExShop.User do
...
+  def changeset_helper(changeset) do
+    changeset
     |> validate_confirmation(:password, message: "password does not match")
     |> validate_format(:email, ~r/@/)
     |> validate_length(:password, min: 5)
     |> unique_constraint(:email)
+    |> set_hashed_password
+  end
+
+  defp set_hashed_password(changeset = %{errors: [_]}), do: changeset
```
Comment (nimish-mehta) [inline apps/ex_shop/web/models/user.ex]: can we pattern match on valid? instead. if in future there are nested assocs here, this won't reflect the errors, also will skip if more than one errors.

File: apps/ex_shop/web/models/user.ex (hunk 2 above)
Comment (pikender) [inline apps/ex_shop/web/models/user.ex:23]: Its added to force and guide that for creation use Registration changeset.
As one might can create one when not aware, and then there will be multiple ways to create user and might not be in sync

Its like a redirection to check registration changeset once, before creating new one

File: apps/ex_shop/web/models/user.ex (hunk 1 above)
Comment (pikender) [inline apps/ex_shop/web/models/user.ex]: Done :+1:

**Pull request closed without merging**